libraries/launcher/org/prismlauncher/fix/skins/SkinFix.java

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2022 TheKodeToad <TheKodeToad@proton.me>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  Linking this library statically or dynamically with other modules is
+ *  making a combined work based on this library. Thus, the terms and
+ *  conditions of the GNU General Public License cover the whole
+ *  combination.
+ *
+ *  As a special exception, the copyright holders of this library give
+ *  you permission to link this library with independent modules to
+ *  produce an executable, regardless of the license terms of these
+ *  independent modules, and to copy and distribute the resulting
+ *  executable under terms of your choice, provided that you also meet,
+ *  for each linked independent module, the terms and conditions of the
+ *  license of that module. An independent module is a module which is
+ *  not derived from or based on this library. If you modify this
+ *  library, you may extend this exception to your version of the
+ *  library, but you are not obliged to do so. If you do not wish to do
+ *  so, delete this exception statement from your version.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.prismlauncher.fix.skins;
+
+import java.net.URL;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+
+import org.prismlauncher.fix.Fix;
+import org.prismlauncher.utils.Base64;
+import org.prismlauncher.utils.Parameters;
+import org.prismlauncher.utils.logging.Log;
+import org.prismlauncher.utils.url.UrlUtils;
+
+/**
+ * Fixes skins by redirecting to other URLs.
+ *
+ * @see {@link Handler}
+ * @see {@link UrlUtils}
+ */
+public final class SkinFix implements Fix, URLStreamHandlerFactory {
+
+    @Override
+    public String getName() {
+        return "legacySkinFix";
+    }
+
+    @Override
+    public boolean isApplicable(Parameters params) {
+        if (!UrlUtils.isSupported() || !Base64.isSupported()) {
+            Log.warning("Cannot access the necessary Java internals for skin fix");
+            Log.warning("Turning off legacy skin fix in Settings > Miscellaneous will silence the warnings");
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public void apply() {
+    	try {
+    		URL.setURLStreamHandlerFactory(this);
```

Comment:
You can drop this import by grabbing a reference to the http default handler before replacing the default factory, like so

```java
    private final Class<? extends URLConnection> defaultHttpConnectionClass;

    @Override
    public void apply() {
    	try {
            URL foo = new URL("http://example.com");
            // Doesn't actually establish a connection
            defaultHttpConnectionClass = foo.openConnection().getClass();
            
            URL.setURLStreamHandlerFactory(this);
    	} catch (Error e) {
            Log.warning("Cannot apply skin fix");
    		Log.warning("URLStreamHandlerFactory is already set");
			Log.warning("Turning off legacy skin fix in Settings > Miscellaneous will silence the warnings");
    	}
    }
    	
    @Override
    public URLStreamHandler createURLStreamHandler(String protocol) {
        if ("http".equals(protocol))
            return new Handler(defaultHttpConnectionClass);

        return null;
    }
```

Reply:
cont.

Then in your handler, like so

```java
        try {
            return defaultHttpConnectionClass.getConstructor(URL.class, Proxy.class).newInstance(url, null);
        } catch (Exception e) {
            // If the constructor isn't found, you can log that out. It's not expected.
            return null;
        }
```